src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -120,36 +120,34 @@ public bool WaitForRequestHandlerConnection(int clientConnectionTimeout)
         return waitSuccess && _handShakeSuccessful;
     }
 
+    public int StartServer()
+    {
+        return _communicationManager.HostServer(new IPEndPoint(IPAddress.Loopback, 0)).Port;
+    }
+
     /// <inheritdoc/>
-    public async Task<int> InitializeCommunicationAsync(int clientConnectionTimeout)
+    public async Task InitializeCommunicationAsync(int clientConnectionTimeout)
     {
         EqtTrace.Info($"VsTestConsoleRequestSender.InitializeCommunicationAsync: Started with client connection timeout {clientConnectionTimeout} milliseconds.");
 
         _processExitCancellationTokenSource = new CancellationTokenSource();
+
         _handShakeSuccessful = false;
         _handShakeComplete.Reset();
-        int port = -1;
+
         try
         {
```

Comment:
This code is completely broken. We are waiting for console to connect here, but we did not return from this method, so the port was never given to the code that starts the console. So there is no process that would ever call us back.